File: ED60/b.rb
if File.exists?("input")
  $stdin = File.open("input")
  $stdout = File.open("output","w")
end

nmk = gets.chomp.split(' ').map(&:to_i)
n = nmk[0]
m = nmk[1]
k = nmk[2]

arr = gets.chomp.split(" ").map(&:to_i)

max=0
maxtwo=0

(0..n-1).each do |i|
	if(arr[i]>max)
    	maxtwo = max
    	max=arr[i]
    elsif(arr[i]>maxtwo)
    	maxtwo = arr[i]
    end
end

ans=(m/(k+1))*(max*(k)+maxtwo)
leftover = m-(m/(k+1))*(k+1);
ans+=max*leftover

puts ans

File: ED60/b.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main(){
	
	#ifndef ONLINE_JUDGE
    	freopen("input", "r", stdin);
    	freopen("output", "w", stdout);
	#endif

    int n,m,k;
    cin>>n>>m>>k;

    int arr[n];
    for(int i=0;i<n;i++){
    	cin>>arr[i];
    }

    int max=0;
    int maxtwo=0;

    for(int i=0;i<n;i++){
    	if(arr[i]>max){
    		maxtwo = max;
    		max=arr[i];
    	}
    	else if(arr[i]>maxtwo){
    		maxtwo = arr[i];
    	}
    }

//    cout<<max<<endl;
//    cout<<maxtwo<<endl;

    ll ans;
    ans=(m/(k+1))*(max*(k)+maxtwo);
//    cout<<k<<endl;

    int leftover = m-(m/(k+1))*(k+1);
    ans+=max*leftover;
    cout<<ans<<endl;
}